fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -5428,8 +5429,7 @@ class DDTeamCollectionUnitTest {
 		 */
 		std::vector<UID> completeSources{ UID(1, 0), UID(2, 0), UID(3, 0) };
 
-		state GetTeamRequest req(
-		    WantNewServers::False, WantTrueBest::True, PreferLowerDiskUtil::True, TeamMustHaveShards::False);
+		state GetTeamRequest req(TeamSelect::WANTSRCSERVERS, PreferLowerDiskUtil::True, TeamMustHaveShards::False);
```

Comment:
Here and the next one (line 5483):  This change actually changes the behavior. Previously, if the `if (!req.wantsNewServers) {}` block (line 253) didn't yield a healthy team, we'd continue into the `if (req.wantsTrueBest) {}` block (line 260). Now, we won't go into the second block. We might need to add another enum case for this situation (maybe: WANT_SRC_OR_TRUE_BEST). We can also discuss this offline.

Reply:
Theoretically, we won't bump into the situation that `req.wantsNewServers = false && req.wantsTrueBest = true`, which means we don't want new servers and we want true best team in the cluster.
The snippet below also shows that:
```
wantsNewServers(isDataMovementForMountainChopper(rs.moveReason) ||
	                                                              isDataMovementForValleyFiller(rs.moveReason) ||
	                                                              rs.moveReason == DataMovementReason::SPLIT_SHARD ||
	                                                              rs.moveReason == DataMovementReason::TEAM_REDUNDANT),

state WantTrueBest wantTrueBest(isValleyFillerPriority(rd.priority));
```

Reply:
And I think we can eliminate this incorrect situation by this method, i.e., using a enum to squash two bool values into one.

Reply:
> Theoretically, we won't bump into the situation that req.wantsNewServers = false && req.wantsTrueBest = true, which means we don't want new servers and we want true best team in the cluster.

Agree with this.